### PR TITLE
Fix sdist editable install and add tests

### PR DESCRIPTION
**Pull request by blink1073**
## References
In the `maintainer-tools` [downstream test action](https://github.com/jupyterlab/maintainer-tools#test-downstream-libraries) we install in editable mode from an sdist by default.  In JupyterLab this was breaking because `scripts/ensure-buildutils.js` was not in the sdist, but really we shouldn't be running node at all.

## Code changes
Cleans up installing from sdist and adds a test for the sdist, as well as a test with the minimum supported dependencies using the new action from `maintainer-tools`.

## User-facing changes
Can now install in editable mode from an sdist.

## Backwards-incompatible changes
None

### CODE REVIEW

File: .github/workflows/linuxtests.yml [hunk 1]
```diff
@@ -76,3 +76,65 @@ jobs:
         with:
           name: ${{ matrix.group }} ${{ github.run_number }}
           path: ./build/${{ matrix.group }}_output
+
+  test_miniumum_verisons:
```
Comment (jtpio) [inline .github/workflows/linuxtests.yml:80]: Minor typo spotted:

```suggestion
  test_minimum_versions:
```

File: setup.cfg [hunk 2]
```diff
@@ -35,10 +35,10 @@ install_requires =
     packaging
     tornado>=6.1.0
     jupyter_core
-    jupyterlab_server~=2.10
-    jupyter_server~=1.4
+    jupyterlab_server>=2.10
+    jupyter_server>=1.8
```
Comment (fcollonval) [inline setup.cfg:39]: Do we want to allow future major versions to be installed with older JupyterLab?

Comment (blink1073) [reply setup.cfg:39]: This comes from the advice in a [blog post](https://iscinumpy.dev/post/bound-version-constraints/) about dependency handling in Python, cf https://github.com/jupyter-server/jupyter_server/issues/699.

Comment (blink1073) [reply setup.cfg]: I take your point that JupyterLab is an application, and tightly coupled to its server, I put the version constraints back in for those.

Comment (fcollonval) [reply setup.cfg]: Thanks @blink1073

**Review by fcollonval — APPROVED** [review]
LGTM

**Pull request merged**